barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/circuit_builder_base.hpp

```diff
@@ -107,9 +107,15 @@ template <typename FF_> class CircuitBuilderBase {
      * @brief `real_variable_tags` is the tagging mechanism for the the multiset-equality check.
      *
      * @details The generalized permutation argument checks both copy constraints and multiset equalities. This is
-     * mediated by a tag; each real variable has a tag. (By default, the tags are set to `DUMMY_TAG == 0`. We assume
-     * that these are not involved in any non-trivial multiset-equality checks.)
-     *
+     * mediated by a tag; each real variable has a tag. (By default, the tags are set to `DUMMY_TAG == 0`.)
+     * @note We make the following IMPORTANT ASSUMPTIONS about the use of tags in circuits.
+     *   * A usual witness being (non-trivially) tagged corresponds to it being range-constrained. Moreover, the tag
+     *   corresponds to the range constraint itself.
+     *   * The derived witness `record` (a.k.a. Reed-Solomon fingerprint), occuring in memory operations, depends on FS
+     *   randomness. These members are tagged to verify the fidelity of the memory operations. (The memory operation
```

Comment:
Note sure if this should be deleted or if there was an additional incomplete thought:
```suggestion
     *   randomness. These members are tagged to verify the fidelity of the memory operations.
```